servers/republik/modules/crowdfundings/lib/AutoPay.js

```diff
@@ -0,0 +1,220 @@
+const debug = require('debug')('crowdfundings:lib:AutoPay')
+
+const { applyPgInterval: { add: addInterval } } = require('@orbiting/backend-modules-utils')
+
+const createCharge = require('./payments/stripe/createCharge')
+const { getCustomPackages } = require('./User')
+const { getLastEndDate } = require('./utils')
+const createCache = require('./cache')
+
+const suggest = async (membershipId, pgdb) => {
+  // Find membership
+  const membership = await pgdb.public.memberships.findOne({ id: membershipId })
+
+  if (!membership) {
+    return false
+  }
+
+  // Find current periods
+  const membershipPeriods = await pgdb.public.membershipPeriods.find({ membershipId: membership.id })
+
+  // Find pledgeOptions
+  const relatedPledgeOptions = await pgdb.public.pledgeOptions.find(
+    {
+      or: [
+        { membershipId },
+        { pledgeId: membership.pledgeId }
+      ]
+    }
+  )
+
+  if (relatedPledgeOptions.length < 1) {
+    return false
+  }
+
+  // Find latest pledge
+  const pledge = await pgdb.public.pledges.findOne(
+    { id: relatedPledgeOptions.map(po => po.pledgeId), status: 'SUCCESSFUL' },
+    { orderBy: { createdAt: 'DESC' }, limit: 1 }
+  )
+
+  // to get membership-related rewardId
+  const membershipTypes = await pgdb.public.membershipTypes.findAll()
+  const membershipPackageOptions = await pgdb.public.packageOptions.find(
+    { rewardId: membershipTypes.map(mt => mt.rewardId) }
+  )
+
+  const pledgeOptions =
+    (await pgdb.public.pledgeOptions.find({
+      pledgeId: pledge.id,
+      'amount >=': 1
+    }))
+      .map(pledgeOption => ({
+        ...pledgeOption,
+        packageOption: membershipPackageOptions.find(po => po.id === pledgeOption.templateId)
+      }))
+
+  const withOtherRewards = pledgeOptions.filter(po => !po.packageOption).length > 1
+  const withOtherMemberships = pledgeOptions.filter(po => !!po.packageOption).length > 1
+  const membershipPledgeOptions = pledgeOptions.filter(po => !!po.packageOption)
+
+  const rewardId = membershipPledgeOptions > 1
+    ? membershipPledgeOptions.find(po => po.membershipId === membershipId).packageOption.rewardId
+    : membershipPledgeOptions[0].packageOption.rewardId
+
+  // Find latest pledge payments
+  const pledgePayments = await pgdb.public.pledgePayments.find({ pledgeId: pledge.id })
+
+  // Find latest pledge payment
+  const payment = await pgdb.public.payments.findOne(
+    { id: pledgePayments.map(p => p.paymentId) },
+    { orderBy: { createdAt: 'DESC' }, limit: 1 }
+  )
+
+  // Pick package and options which may be used to submit and autopayment
+  const user = await pgdb.public.users.findOne({ id: membership.userId })
```

Comment:
(strictly speaking this user should be transformed for safety and to be consistent with the usage of `crowdfundings/graphql/resolvers/User.js` but since in the end mostly user.id is used and this object never gets exposed it probably does not matter. However if there is downstream code that would relay on _raw it would explode)